packages/core/__tests__/language-server/requests.test.ts

```diff
@@ -0,0 +1,328 @@
+        },
+      },
+      {
+        newText: '',
+        range: {
+          start: { character: 0, line: 1 },
+          end: { character: 0, line: 2 },
+        },
+      },
+      {
+        newText: '',
+        range: {
+          start: { character: 0, line: 2 },
+          end: { character: 0, line: 3 },
+        },
+      },
+      {
+        newText: '',
+        range: {
+          start: { character: 0, line: 3 },
+          end: { character: 0, line: 4 },
+        },
+      },
+      {
+        newText: '',
+        range: {
+          start: { character: 0, line: 4 },
+          end: { character: 0, line: 5 },
+        },
+      },
+    ]);
+
+    test('many imports, sorted', () => {
+      project.write({
+        'index.ts': stripIndent`
+        import Component from '@ember/component';
+        import { ComponentLike, WithBoundArgs } from '@glint/template';
+        import EmberComponent from './ember-component';
+        import './App.css';
+        import logo from './logo.svg';
+
+          interface WrapperComponentSignature {
+            Blocks: {
+              default: [
+                {
+                  InnerComponent: WithBoundArgs<typeof EmberComponent, 'required'>;
+                  MaybeComponent?: ComponentLike<{ Args: { key: string } }>;
+                }
+              ];
+            };
+          }
+
+          export default class WrapperComponent extends Component<WrapperComponentSignature> {
+            logo = logo
+          }
+        `,
+      });
+
+      let server = project.startLanguageServer();
+
+      let formatting = ts.getDefaultFormatCodeSettings();
+      let preferences = {};
+      let edits = server.organizeImports(project.fileURI('index.ts'), formatting, preferences);
+
+      expect(edits).toEqual([
+        {
+          newText:
+            "import Component from '@ember/component';\nimport { ComponentLike, WithBoundArgs } from '@glint/template';\nimport './App.css';\nimport EmberComponent from './ember-component';\nimport logo from './logo.svg';\n",
+          range: {
+            start: { character: 0, line: 0 },
+            end: { character: 0, line: 1 },
+          },
+        },
+        {
+          newText: '',
+          range: {
+            start: { character: 0, line: 1 },
+            end: { character: 0, line: 2 },
+          },
+        },
+        {
+          newText: '',
+          range: {
+            start: { character: 0, line: 2 },
+            end: { character: 0, line: 3 },
+          },
+        },
+        {
+          newText: '',
+          range: {
+            start: { character: 0, line: 3 },
+            end: { character: 0, line: 4 },
+          },
+        },
+        {
+          newText: '',
+          range: {
+            start: { character: 0, line: 4 },
+            end: { character: 0, line: 5 },
+          },
+        },
+      ]);
+    });
+  });
+
+  test('many imports, already sorted', () => {
```

Comment:
How many of these tests are exercising different code paths in Glint, versus different code paths within TypeScript itself? This is the opposite of the question I usually find myself asking in code reviews, but do you think each of these tests is providing unique value? 🙂

Reply:
Ha, fair enough there's probably some duplication in what's being covered. I'll take another look at the suite